internal/executor/doctor/executor.go

```diff
@@ -178,29 +158,53 @@ func (d *Executor) Help(context.Context) (api.Message, error) {
 	}, nil
 }
 
-func (d *Executor) getGptClient(cfg *Config) (gpt3.Client, error) {
+func (d *Executor) getGptClient(log logrus.FieldLogger, cfg Config) (gpt3.Client, error) {
 	d.l.Lock()
 	defer d.l.Unlock()
+
+	if d.gptClient != nil {
+		return d.gptClient, nil
+	}
+
 	if cfg.APIKey == "" {
-		return nil, fmt.Errorf("OpenAPI API Key cannot be empty. You generate it here: https://platform.openai.com/account/api-keys")
+		return nil, fmt.Errorf("API Key cannot be empty. If you use OpenAI API, generate it here: https://platform.openai.com/account/api-keys")
 	}
 
-	var opts []gpt3.ClientOption
+	baseURL := defaultAPIBaseURL
 	if cfg.APIBaseURL != "" {
-		opts = append(opts, gpt3.WithBaseURL(cfg.APIBaseURL))
+		baseURL = cfg.APIBaseURL
 	}
 
+	defaultEngine := gpt3.TextDavinci003Engine
 	if cfg.DefaultEngine != "" {
-		opts = append(opts, gpt3.WithDefaultEngine(cfg.DefaultEngine))
+		defaultEngine = cfg.DefaultEngine
 	}
 
+	userAgent := defaultUserAgent
+	if cfg.UserAgent != "" {
+		userAgent = cfg.UserAgent
+	}
+
+	orgID := ""
 	if cfg.OrganizationID != "" {
-		opts = append(opts, gpt3.WithOrg(cfg.OrganizationID))
+		orgID = cfg.OrganizationID
 	}
 
-	if d.gptClient == nil {
-		d.gptClient = gpt3.NewClient(cfg.APIKey, opts...)
+	opts := []gpt3.ClientOption{
+		gpt3.WithBaseURL(baseURL),
+		gpt3.WithDefaultEngine(defaultEngine),
+		gpt3.WithUserAgent(userAgent),
+		gpt3.WithOrg(orgID),
 	}
+
+	log.WithFields(logrus.Fields{
+		"baseURL":       baseURL,
+		"defaultEngine": defaultEngine,
+		"userAgent":     userAgent,
+		"orgID":         orgID,
+	}).Debugf("Creating GPT3 client with API key starting with %q...", stringsutil.ShortenString(cfg.APIKey, printAPIKeyCharCount))
```

Comment:
```suggestion
		"apiKeyPrefix":         stringsutil.ShortenString(cfg.APIKey, printAPIKeyCharCount),
	}).Debugf("Creating GPT3 client...")
```

would be better to keep the same style, otherwise, you could do:
`Creating GPT3 client with API key starting with %q for endpoint %q using %q engine in the context of %q org id` 😄

Reply:
Yes and no, but ok 😄